DAQ/src/StrawDigisFromArtdaqFragments_module.cc

```diff
@@ -0,0 +1,437 @@
+              int mn_id    = hit_data->StrawIndex >> 7;
+              int panel_id = panelID(mn_id,dtc_id,link_id);
+              if (panel_id < 0) {
+                print_(std::format("ERROR: hit chid:{:04x} inconsistent with the dtc_id:{} and link_id:{}",
+                                   hit_data->StrawIndex, dtc_id, link_id));
+//-----------------------------------------------------------------------------
+// in case of a single channel ID error no need to skip the rest of the ROC data -
+// force geographical address and mark the produced digi
+//-----------------------------------------------------------------------------
+                panel_id  = panel_map_[dtc_id][link_id];
+                digi_flag = mu2e::StrawDigiFlag::corrupted;
+              }
+              else if (hit_data->NumADCPackets != nADCPackets_) {
+                int np = hit_data->NumADCPackets;
+                print_(std::format("ERROR: wrong NADCpackets:{} , expected:{}, STOP PROCESSING HITS",
+                                   np,nADCPackets_));
+                break;
+              }
+//-----------------------------------------------------------------------------
+// convert channel_id into a strawID
+//-----------------------------------------------------------------------------
+              uint16_t straw_index = (panel_id << 7) | ch_id;
+              mu2e::StrawId sid(straw_index);
+              mu2e::TrkTypes::TDCValues tdc = {hit_data->TDC0(), hit_data->TDC1()};
+              mu2e::TrkTypes::TOTValues tot = {hit_data->TOT0, hit_data->TOT1};
+              mu2e::TrkTypes::ADCValue  pmp = hit_data->PMP;
+
+              print_(std::format("offset:0x{:04x} sid:{:5} times: {:9} {:9} TOT:{:2}:{:2} pmp:{}",
+                                 offset, straw_index,hit_data->TDC0(),hit_data->TDC1(),tot[0],tot[1],pmp),1);
+
+              straw_digis->emplace_back(sid, tdc, tot, pmp);
+//-----------------------------------------------------------------------------
+// an if could be more disruptive
+//-----------------------------------------------------------------------------
+              auto digi = straw_digis->back();
+              digi.digiFlag() = digi_flag;
+//------------------------------------------------------------------------------
+// the corresponding waveform
+//-----------------------------------------------------------------------------
+              if (saveWaveforms_) {
+                std::vector<uint16_t> wf(nSamples_);
+
+                wf[0] = hit_data->ADC00;
+                wf[1] = hit_data->ADC01();
+                wf[2] = hit_data->ADC02;
+
+                auto npackets   = 0;
+                auto idx        = 2;
+                auto adc_packet = (mu2e::TrackerDataDecoder::TrackerADCPacket*)((char*) hit_data + 16); // the packet size is 16 bytes
+                while (npackets < nADCPackets_) {
+                  wf[++idx] = adc_packet->ADC0;
```

Comment:
I am confused that some adc_packet members are data members and some are functions. This is outside scope of this PR, but shouldn't we consolidate to a single access pattern?